client/landing/stepper/declarative-flow/internals/steps-repository/unified-plans/index.tsx

```diff
@@ -80,6 +92,39 @@ export default function PlansStepAdaptor( props: StepProps ) {
 		setPlanInterval( intervalType );
 	};
 
+	/**
+	 *  Plan step switches the selected theme to default twentytwentyfour when the plan is free
+	 *  but the selected design requires a paid plan.
+	 */
+	const switchPaidDesignToDefault = ( stepInfo: ProvidedDependencies ) => {
+		const hasPaidDesign = Boolean(
+			selectedDesign?.design_tier && selectedDesign?.design_tier !== 'free'
+		);
+		const isOnboarding = isOnboardingFlow( props.flow ) && isGoalFirstExperiment;
+
+		if ( ! hasPaidDesign || !! stepInfo.cartItems || ! isOnboarding ) {
+			return;
+		}
+
+		if ( site ) {
+			setPendingAction( async () => {
+				return setDesignOnSite( site?.ID, defaultDesign, {
+					styleVariation: defaultDesign?.style_variations?.[ 0 ],
+				} ).then( async ( theme: ActiveTheme ) => {
+					await updateLaunchpadSettings( site?.ID || '', {
+						checklist_statuses: { design_completed: false },
+					} );
+					return reduxDispatch( setActiveTheme( site?.ID || -1, theme ) );
+				} );
+			} );
+		}
+		setSelectedDesign( { ...defaultDesign, default: true } );
```

Comment:
I think we just need to set the selected design to null to avoid changing the design after the site is created 🤔

Reply:
I'm not sure I understand the comment. We are switching the previously selected design to the default theme at this point.

Reply:
The previously selected design won't be applied immediately so the site is still using the default theme on the creation.

Reply:
I see what you're saying. I used this design to determine whether to show the design_selected task further in the flow. We should be able to change that after we fix the BE so that design_selected is not always true.